Kudu.Services/Diagnostics/ProfileManager.cs

```diff
@@ -0,0 +1,314 @@
+                    catch
+                    {
+                    }
+                }
+
+                DisposeTimerIfNecessary();
+
+                return new HttpResponseInfo(HttpStatusCode.OK, string.Empty);
+            }
+        }
+
+        private static async Task<HttpResponseInfo> ExecuteProfilingCommandAsync(string arguments, ITracer tracer)
+        {
+            MemoryStream outputStream = null;
+            MemoryStream errorStream = null;
+            try
+            {
+                tracer.Step("ProcessName:" + _processName + "   arguments:" + arguments);
+                var exe = new Executable(_processName, Path.GetDirectoryName(_processName), TimeSpan.FromSeconds(ProcessExitTimeoutInSeconds));
+
+                outputStream = new MemoryStream();
+                errorStream = new MemoryStream();
+
+                tracer.Step("Path:" + exe.Path + " working directory:" + exe.WorkingDirectory);
+
+                int exitCode = await exe.ExecuteAsync(tracer, arguments, outputStream, errorStream);
+
+                string output = GetString(outputStream);
+                string error = GetString(errorStream);
+
+                tracer.Step(output);
+
+                if (exitCode != 0)
+                {
+                    tracer.TraceError(string.Format(CultureInfo.InvariantCulture, "Starting process {0} failed with the following error code '{1}'.", _processName, exitCode));
+                    return new HttpResponseInfo(HttpStatusCode.InternalServerError, "Profiling process failed with the following error code: " + exitCode);
+                }
+                else if (!string.IsNullOrEmpty(error))
+                {
+                    tracer.TraceError(error);
+                    return new HttpResponseInfo(HttpStatusCode.InternalServerError, "Profiling process failed with the following error: " + error);
+                }
+
+                return new HttpResponseInfo(HttpStatusCode.OK, string.Empty);
+            }
+            catch (Exception ex)
+            {
+                tracer.TraceError(ex);
+                return new HttpResponseInfo(HttpStatusCode.InternalServerError, ex.Message);
+            }
+            finally
+            {
+                if (outputStream != null)
+                {
+                    outputStream.Dispose();
+                }
+
+                if (errorStream != null)
+                {
+                    errorStream.Dispose();
+                }
+            }
+        }
+
+        private static int GetNextProfilingSessionId()
+        {
+            // TODO: This is not a good way to track active profiling sessions, but as of VS2015 RTM, the profiling service does not provide any API to track the current active sessions.  
+            // This is planned to be fixed in VS2015 Update 1.
+            var r = new Random();
+
+            int sessionId = r.Next(1, 255);
+            List<ProfileInfo> list = _profilingList.Values.ToList<ProfileInfo>();
```

Comment:
I just don't understand the use of random numbers here. Why not simply increment a number to get sequential IDs?